src/auth/оAuth2.ts

```diff
@@ -233,13 +233,15 @@ export default class OAuth2 {
             this.setCredentials(refreshedToken);
 
             this.emitter.emit('refreshAuthToken', refreshedToken);
+
+            return refreshedToken;
         } catch (ex) {
             log('Failed to refresh the token', ex);
             throw ex;
         }
     }
 
-    public async refreshToken() {
+    public async refreshToken(): Promise<Token> {
```

Comment:
If clientToken is refreshend, string will be returned. The correct type should be "Promise<Token | string>

Reply:
Oh, why will a string be returned? The return type of `refreshClientToken` is `Promise<Token>`, and it looks like it does actually return a token - for example, it calls `this.setClientToken(token);` on the returned `token`, and that function expects a `Token`.

Reply:
Nevermind, I was looking in wrong part of the code :)